packages/rocketchat-api/server/default/info.js

```diff
@@ -1,5 +1,13 @@
 RocketChat.API.default.addRoute('info', { authRequired: false }, {
 	get: function() {
-		return RocketChat.Info;
+		if (this.request.headers['x-user-id'] != null && RocketChat.authz.hasRole(this.request.headers['x-user-id'], 'admin')) {
```

Comment:
You are correct, but that's why I said a helper method as we have added functionality on top of Restivus which allows adding functions (helper methods) to the endpoint context. Then in that helper method you would access the user's auth token and user ID in the headers to then authenticate with Meteor which is actually kind of done on the `api.js` file inside if you need an example.